frontend/src/components/employees/index.tsx

```diff
@@ -136,9 +141,65 @@ const addButton = (
   </div>
 );
 
+const deleteModalBody = (
+  staff: StaffData,
+  closeModalFunction: () => void,
+  deleteCallback: CallbackFunction
+) => (
+  <div className={employeesStyle.modalDeleteContent}>
+    <div className={employeesStyle.modalDeleteHeader}>
+      <h4>Delete</h4>
+      <button
+        type="button"
+        aria-label="closeButton"
+        className={employeesStyle.xButton}
+        onClick={closeModalFunction}>
+        <Clear />
+      </button>
+    </div>
+    <hr />
+    <span>
+      Are you sure you want to delete {staff.firstName} {staff.surname}?
+    </span>
+    <div className={employeesStyle.modalDeleteButtons}>
+      <Button variant="contained" onClick={closeModalFunction}>
+        Cancel
+      </Button>
+      <Button
+        variant="contained"
+        color="secondary"
+        onClick={deleteCallback(staff)}>
+        DELETE
+      </Button>
+    </div>
+  </div>
+);
+
 const Employees = () => {
   const staffList = useSelector(selectStaff);
   const dispatch = useDispatch();
+  const [isModalOpen, setModalState] = useState(false);
+  const [modalBody, setModalBody] = useState(<div />);
+
```

Comment:
is setting HTML elements in state a normal thing to do? My understanding was that state should aim to contain the minimum information possible to tell you (along with props) what the component should look like.

Reply:
I don't know what's normal, but in this case the modal needs to know which button in the table was clicked and this is the first solution that came to mind - setting the body in the callback function.